extensions/exchanges/bitstamp/exchange.js

```diff
@@ -0,0 +1,326 @@
+    name: 'bitstamp',
+    historyScan: false,
+    makerFee: 0.25,
+    takerFee: 0.25,
+
+    getProducts: function () {
+      return require('./products.json')
+    },
+
+    //-----------------------------------------------------
+    // Public API functions
+    // getQuote() and getTrades() are using Bitstamp websockets
+    // The data is not done by calling the interface function,
+    // but rather pulled from the "wstrades" and "wsquotes" JSOM objects
+    // Those objects are populated by the websockets event handlers
+
+    getTrades: function (opts, cb) {
+      var args = {
+        wait: 2,   // Seconds
+        product_id: wsOpts.currencyPair
+      }
+      if (typeof wstrades.time == undefined) return retry('getTrades', args)
+      var t = wstrades
+      var trades = t.map(function (trade) {
+        return (trade)
+      })
+      cb(null, trades)
+    },
+
+    getQuote: function (opts, cb) {
+      var args = {
+        wait: 2,   // Seconds
+        currencyPair: wsOpts.currencyPair
+      }
+      if (typeof wsquotes.bid == undefined) return retry('getQuote', args )
+      cb(null, wsquotes)
+    },
+
+    //-----------------------------------------------------
+    // Private (authenticated) functions
+    //
+
+    getBalance: function (opts, cb) {
+      var client = authedClient()
+      client.balance(null, function (err, body) {
+        body = statusErr(err,body)
+        var balance = {asset: 0, currency: 0}
+        balance.currency = body[opts.currency.toLowerCase() + '_available']
+        balance.asset = body[opts.asset.toLowerCase() + '_available']
+        balance.currency_hold = 0
+        balance.asset_hold = 0
+        cb(null, balance)
+      })
+    },
+
+    cancelOrder: function (opts, cb) {
+      var func_args = [].slice.call(arguments)
+      var client = authedClient()
+      client.cancel_order(opts.order_id, function (err, body) {
+        body = statusErr(err,body)
+	if (body.status === 'error') {
+	  return retry('cancelOrder', func_args, err)
+	}
+        cb()
+      })
+    },
+
+    trade: function (type,opts, cb) {
+//console.log('Trade ' + type + ' options:\n',opts)
+      var client = authedClient()
+      var currencyPair = joinProduct(opts.product_id).toLowerCase()
+      if (typeof opts.order_type === 'undefined' ) {
+        opts.order_type = 'maker'
+      }
+      // Bitstamp has no "post only" trade type
+      opts.post_only = false
+//opts.order_type = 'taker'
+      if (opts.order_type === 'maker') {
+        client.tradeDaily(type, currencyPair, opts.size, opts.price, function (err, body) {
+          body = statusErr(err,body)
+	  if (body.status === 'error') {
+	    var order = { status: 'rejected', reject_reason: 'balance' }
+            return cb(null, order)
+	  } else { 
+	    // Statuses:
+	    // 'In Queue', 'Open', 'Finished'
+	    body.status = 'done'
+	  }
+          orders['~' + body.id] = body
+          cb(null, body)
+        })
+      } else {
```

Comment:
perhaps we can be more explicit here and just say
`if (opts.order_type === 'taker') {`